drivers/rtc/rtc_smartbond.c

```diff
@@ -0,0 +1,632 @@
+	data->alarm_user_data = user_data;
+
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+
+static int rtc_smartbond_alarm_get_supported_fields(const struct device *dev, uint16_t id,
+						uint16_t *mask)
+{
+	if (id != RTC_ALARM_VALID_ID) {
+		LOG_ERR("Alarm id is out of range");
+		return -EINVAL;
+	}
+
+	if (mask == NULL) {
+		LOG_ERR("Pointer to store the mask value is missed");
+		return -EINVAL;
+	}
+
+	*mask = (uint16_t)RTC_SMARTBOND_SUPPORTED_ALARM_FIELDS;
+
+	return 0;
+}
+#endif
+
+#if defined(CONFIG_RTC_UPDATE)
+static int rtc_smartbond_update_set_callback(const struct device *dev, rtc_update_callback callback,
+						void *user_data)
+{
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	data->update_cb = callback;
+	data->update_user_data = user_data;
+
+	if (data->update_cb) {
+		/* Enable asserting the RTC interrupt line when the second counter rolls over. */
+		RTC->RTC_INTERRUPT_ENABLE_REG = RTC_RTC_INTERRUPT_ENABLE_REG_RTC_SEC_INT_EN_Msk;
+	} else {
+		RTC->RTC_INTERRUPT_DISABLE_REG = RTC_RTC_INTERRUPT_DISABLE_REG_RTC_SEC_INT_DIS_Msk;
+	}
+
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+#endif
+
+struct rtc_driver_api rtc_smartbond_driver_api = {
+	.get_time = rtc_smartbond_get_time,
+	.set_time = rtc_smartbond_set_time,
+#if defined(CONFIG_RTC_ALARM)
+	.alarm_get_time = rtc_smartbond_alarm_get_time,
+	.alarm_set_time = rtc_smartbond_alarm_set_time,
+	.alarm_is_pending = rtc_smartbond_alarm_is_pending,
+	.alarm_set_callback = rtc_smartbond_alarm_set_callback,
+	.alarm_get_supported_fields = rtc_smartbond_alarm_get_supported_fields,
+#endif
+#if defined(CONFIG_RTC_UPDATE)
+	.update_set_callback = rtc_smartbond_update_set_callback,
+#endif
+#if defined(CONFIG_RTC_CALIBRATION)
+	.set_calibration = NULL,
+	.get_calibration = NULL,
+#endif
+};
+
+static void rtc_smartbond_100HZ_clock_cfg(void)
+{
+#define RTC_DIV_DENOM_1000  0
+#define RTC_DIV_DENOM_1024  1
+
+	const struct device * const dev = DEVICE_DT_GET(DT_NODELABEL(osc));
+	uint32_t lp_clk_rate;
+	uint32_t clk_rtcdiv_reg;
+
+	if (!device_is_ready(dev)) {
+		__ASSERT_MSG_INFO("Clock device is not ready");
+	}
+
+	if (clock_control_get_rate(dev, (clock_control_subsys_t)SMARTBOND_CLK_LP_CLK,
+								&lp_clk_rate) < 0) {
+		__ASSERT_MSG_INFO("Cannot extract LP clock rate");
+	}
+
+	clk_rtcdiv_reg = CRG_TOP->CLK_RTCDIV_REG;
+	CLK_RTCDIV_REG_SET_FIELD(DENOM, clk_rtcdiv_reg, RTC_DIV_DENOM_1000);
+	CLK_RTCDIV_REG_SET_FIELD(INT, clk_rtcdiv_reg, lp_clk_rate / 100);
+	CLK_RTCDIV_REG_SET_FIELD(FRAC, clk_rtcdiv_reg, (lp_clk_rate % 100) * 10);
+	CRG_TOP->CLK_RTCDIV_REG = clk_rtcdiv_reg;
+}
+
+static int rtc_smartbond_init(const struct device *dev)
+{
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	memset(data, 0, sizeof(*data));
+
+	/* Wakeup device from RTC events (alarm/roll over) */
+#if CONFIG_PM
+	bool is_xtal32m_enabled = !!DT_NODE_HAS_STATUS(DT_NODELABEL(xtal32m), okay);
```

Comment:
don't think the `!!` is neede, it's a bool, the compiler should do the right thing automatically

Reply:
Fixed